src/main/java/nextstep/subway/path/domain/Path.java

```diff
@@ -7,39 +7,50 @@
 import org.jgrapht.graph.DefaultWeightedEdge;
 import org.jgrapht.graph.WeightedMultigraph;
 
+import nextstep.subway.auth.domain.LoginMember;
 import nextstep.subway.common.ErrorCode;
 import nextstep.subway.line.domain.Distance;
 import nextstep.subway.line.domain.Line;
+import nextstep.subway.line.domain.Section;
 import nextstep.subway.path.exception.PathException;
 import nextstep.subway.station.domain.Station;
 
 public class Path {
 
 	private List<Station> stations;
 	private Distance distance;
+	private Price price;
 
 	protected Path() {
 	}
 
-	private Path(final List<Station> stations, final Distance distance) {
+	private Path(final List<Station> stations, final Distance distance, final Price price) {
 		this.stations = stations;
 		this.distance = distance;
+		this.price = price;
 	}
 
 	public static Path of(List<Line> lines, Station sourceStation, Station targetStation) {
 		validSourceTargetStation(sourceStation, targetStation);
-		GraphPath<Station, DefaultWeightedEdge> shortestPath = generateShortestPath(lines, sourceStation,
+		GraphPath<Station, LineWeightedEdge> shortestPath = generateShortestPath(lines, sourceStation,
 			targetStation);
+		Distance distance = Distance.from((int)shortestPath.getWeight());
 
-		return new Path(shortestPath.getVertexList(), Distance.from((int)shortestPath.getWeight()));
+		return new Path(shortestPath.getVertexList(), distance, Price.of(distance, shortestPath.getEdgeList()));
 	}
 
-	public List<Station> getStations() {
-		return stations;
-	}
+	public static Path of(List<Line> lines, Station sourceStation, Station targetStation, LoginMember loginMember) {
+		GraphPath<Station, LineWeightedEdge> shortestPath = generateShortestPath(lines, sourceStation,
+			targetStation);
+		validSourceTargetStation(sourceStation, targetStation);
+		Distance distance = Distance.from((int)shortestPath.getWeight());
```

Comment:
랩핑 객체를 사용하는곳에서 타입에 맞춰서 생성인자를 전달하는것도 좋지만, 객체를 더 유연하게 사용할 수 있도록 여러 형태의 생성자를 열어두는것도 좋을 것 같습니다.

```suggestion
		Distance distance = Distance.from(shortestPath.getWeight());
```
```java
	// Distance 클래스
	public static Distance from(double distance) {
		return new Distance((int) distance);
	}
```